### PR TITLE
[shaderwriter] Add new port.

### LINKED ISSUE #13894: [New Port Request] ShaderWriter
Library name: ShaderWriter

Library description: Allows writing shaders directly from C++ code, supports GLSL, HLSL and SPIRV outputs.

Source repository URL: https://github.com/DragonJoker/ShaderWriter

Anything else that is useful to know when adding (such as optional features the library may have that should be included):
- Supported on osx, linux, windows
- Compiled on x86, x64, arm, and arm64

### PR DESCRIPTION
**Pull request by DragonJoker**
**Describe the pull request**

* What does your PR fix?
  Fixes #13894  .

* Which triplets are supported/not supported? Have you updated the CI baseline?
  I have tested `x64-windows`, `x64-windows-static`, `x86-windows`, `x64-linux` and `x64-osx`, and I expect the other ones to build as well.
  The CI baseline has not been updated.

* Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes.

### CODE REVIEW

File: ports/shaderwriter/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
+find_program(GIT git)
+
+set(GIT_URL "https://github.com/DragonJoker/ShaderWriter.git")
+set(GIT_BRANCH "master")
+set(GIT_REV "e978c96c959e8aa41eedaef322dcc6a0ec00ad49")
+set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/${PORT}-${GIT_REV})
+
+if(NOT EXISTS "${SOURCE_PATH}/.git")
```
Comment (LilyWangL) [inline ports/shaderwriter/portfile.cmake:9]: I think you can use [vcpkg_from_github](https://github.com/microsoft/vcpkg/blob/ae74e0bfd37c3a0a0a096acfc0cf0851582b569d/docs/maintainers/vcpkg_from_github.md#vcpkg_from_github) here.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/shaderwriter/portfile.cmake (hunk 1 above)
Comment (DragonJoker) [reply ports/shaderwriter/portfile.cmake:9]: I've had some issues with my submodule, when I tried it at first, but I can give it another try.

Comment (DragonJoker) [reply ports/shaderwriter/portfile.cmake:9]: With the following snippet:
```cmake
vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
    REPO DragonJoker/ShaderWriter
    REF e978c96c959e8aa41eedaef322dcc6a0ec00ad49
    HEAD_REF master
    SHA512 4f1ea1fd97392edc2b3dd9b6453fa395772e75a52f28d7bbd5a63d2fe8eca40c8a0d4016e2017e126f22fc03acf28ab29db6de037d64c7687f95884b8435b7c7
)
find_program(GIT git REQUIRED)
vcpkg_execute_required_process(
    COMMAND ${GIT} submodule update --init CMake
    WORKING_DIRECTORY ${SOURCE_PATH}
    LOGNAME submodule
)
#
# ...
#
```

I get the following error:
`error: pathspec 'CMake' did not match any file(s) known to git`

So, if you have any solution, I'm open to it ^^

Comment (Hoikas) [reply ports/shaderwriter/portfile.cmake:9]: I started a review with this same suggestion, but I realized that `vcpkg_from_github` was downloading the repo archive, which IIRC does not play well with submodules and lacks the .git directory. So, I wonder if the original solution is the best way to proceed.

File: ports/shaderwriter/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "shaderwriter",
+  "version-string": "0.1.0",
```
Comment (LilyWangL) [inline ports/shaderwriter/vcpkg.json:3]: Here should use date instead of version number, because it is not a release version.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/shaderwriter/vcpkg.json (hunk 2 above)
Comment (DragonJoker) [reply ports/shaderwriter/vcpkg.json]: Done

**Review by LilyWangL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**